src/rendering/MaterialParser/ConfigLoader.cc

```diff
@@ -0,0 +1,395 @@
+{
+  for (auto & i : m_scriptList) {
+    delete i.second;
+  }
+  m_scriptList.clear();
+}
+
+ConfigNode * ConfigLoader::getConfigScript(const std::string & name)
+{
+  auto i = m_scriptList.find(name);
+  // If found..
+  if (i != m_scriptList.end()) {
+    return i->second;
+  } else {
+    return nullptr;
+  }
+}
+
+std::map<std::string, ConfigNode *> ConfigLoader::getAllConfigScripts()
+{
+  return m_scriptList;
+}
+
+void ConfigLoader::parseScript(std::ifstream & stream)
+{
+  // Get first token
+  _nextToken(stream);
+  if (tok == TOKEN_EOF) {
+    stream.close();
+    return;
+  }
+
+  // Parse the script
+  _parseNodes(stream, 0);
+
+  stream.close();
+}
+
+void ConfigLoader::_nextToken(std::ifstream & stream)
+{
+  lastTok = tok;
+  lastTokVal = tokVal;
+
+  // EOF token
+  if (stream.eof()) {
+    tok = TOKEN_EOF;
+    return;
+  }
+
+  // Get next character
+  int ch = stream.get();
+  if (ch == -1) {
+    tok = TOKEN_EOF;
+    return;
+  }
+  // Skip leading spaces / tabs
+  while ((ch == ' ' || ch == 9) && !stream.eof()) {
+    ch = stream.get();
+  }
+
+  if (stream.eof()) {
+    tok = TOKEN_EOF;
+    return;
+  }
+
+  // Newline token
+  if (ch == '\r' || ch == '\n') {
+    do{
+      ch = stream.get();
+    } while ((ch == '\r' || ch == '\n') && !stream.eof());
+
+    stream.unget();
+
+    tok = TOKEN_NewLine;
+    return;
+  } else if (ch == '{') {
+    // Open brace token
+    tok = TOKEN_OpenBrace;
+    return;
+  } else if (ch == '}') {
+    // Close brace token
+    tok = TOKEN_CloseBrace;
+    return;
+  }
+
+  // Text token
+  if (ch < 32 || ch > 122) {  // Verify valid char
+    throw std::runtime_error(
+      "Parse Error: Invalid character, ConfigLoader::load()");
+  }
+
+  tokVal = "";
+  tok = TOKEN_Text;
+  do{
+    // Skip comments
+    if (ch == '/') {
+      int ch2 = stream.peek();
+
+      // C++ style comment (//)
+      if (ch2 == '/') {
+        stream.get();
+        do{
+          ch = stream.get();
+        } while (ch != '\r' && ch != '\n' && !stream.eof());
+
+        tok = TOKEN_NewLine;
+        return;
+      } else if (ch2 == '*') {
+        stream.get();
+        do{
+          ch = stream.get();
+          ch2 = stream.peek();
+        } while (!(ch == '*' && ch2 == '/') && !stream.eof());
+        stream.get();
+        do{
+          ch = stream.get();
+        } while (ch != '\r' && ch != '\n' && !stream.eof());
+        continue;
+      }
+    }
+
+    // Add valid char to tokVal
+    tokVal += static_cast<char>(ch);
+
+    // Next char
+    ch = stream.get();
+  } while (ch > 32 && ch <= 122 && !stream.eof());
+
+  stream.unget();
+
+  return;
+}
+
+void ConfigLoader::_skipNewLines(std::ifstream & stream)
+{
+  while (tok == TOKEN_NewLine) {
+    _nextToken(stream);
+  }
+}
+
+void ConfigLoader::_parseNodes(std::ifstream & stream, ConfigNode * parent)
+{
+  typedef std::pair < std::string, ConfigNode * > ScriptItem;
+
+  while (true) {
+    switch (tok) {
```

Comment:
minor compile warning here
```
ConfigLoader.cc:206:12: warning: switch missing default case [-Wswitch-default]
```

I can think you can add an empty `default` case that just `break`s.